### DISCUSSION TITLE
Inconsistent Reference Syntax (Proposal)

### DISCUSSION THREAD

**Discussion by setanarut** [godotengine/godot-proposals#4116] [General Discussions]
**Issue description:**
In-editor reference pages and https://docs.godotengine.org use different syntaxes. docs.godotengine.org pages are formatted according to C++ syntax, not GDscript syntax. In Gdscript syntax, type definitions comes after the variable name(`a: int`). In documents, it comes first.(`int a`) The return value type also does not conform to the Gdscript syntax.

**URL to the documentation page (if already existing):**

https://docs.godotengine.org/en/latest/classes/class_node2d.html
<img width="610" alt="Ekran Resmi 2022-02-25 13 38 52" src="https://user-images.githubusercontent.com/36481442/155701965-21ad6ae3-2fc5-4676-b756-0581004a306a.png">

On pages in the editor (screenshot below) for example, `apply_scale()` should be written as;

```gdscript
apply_scale(ratio: Vector2) -> void` ## Gdscript and Rust language style return type (docs.rs).
```

```gdscript
apply_scale(ratio: Vector2) void` ## Go language style return type (pkg.go.dev).
```

<img width="705" alt="Ekran Resmi 2022-02-25 13 39 06" src="https://user-images.githubusercontent.com/36481442/155701885-0702a1bd-6a38-438e-8100-79a8a9cd932a.png">

For those used to the Gdscript, Golang and Rust syntax, the return value is perceived as const.
I always keep my eye on the right side for the return type because I'm used to the Go language.

## Go and Rust screenshots

<img width="938" alt="Ekran Resmi 2022-02-25 17 36 21" src="https://user-images.githubusercontent.com/36481442/155738902-c32b6b85-ef20-450a-8369-b31550365000.png">

<img width="684" alt="Ekran Resmi 2022-02-25 18 10 40" src="https://user-images.githubusercontent.com/36481442/155739061-6da26a20-d793-40f5-8e2e-0a3401f8553f.png">

## Summary

Document formats must conform to the GDscript syntax. Return type should be on the right.

> Note: Previously opened issue page. -> https://github.com/godotengine/godot-docs/issues/5645

**Comment by azur-wolve** [comment]
You're right, but it still enough intuitive.
Only by looking into it can know what every arg-type is.
Maybe adding a first row to the table with:
` | return type | function & args |`
Could solve it.
